### PR TITLE
Tesla Update

### PR DESCRIPTION
**Pull request by Fox-McCloud**
From TG.

Updates the Tesla to the most recent iteration. Overall, it's far far far more deadly than ti was before.

Major Changes
- Tesla can now damage blobs (yes, releasing the Tesla can be a viable way to combat blobs in the ultra late stage now...provided admins give you permission)
- Tesla will now cause explosions when it overloads machines. Needless to say, this makes the Tesla an incredibly deadly if it ever escapes.
- Tesla's energy will dissipate more slowly and charge up more quickly
- The Tesla moving near a mob will now dust it (as opposed to the mob having to move into the Tesla)
  -  Being near a grounding rod prevents you from being dusted from the Tesla

Related
- Can now pulse/attach assemblies to Tesla coils---set up an engineer defense or autism fort that's protected by the ***power of electricity***

Code stuffs
- Cleaned up singularity and Tesla code a bit here and there
- Various performance tweaks such as utilize `urange` or z-level checks for loops through lists

Mandatory Picture of Tesla Aftermath
img

:cl: Fox McCloud
tweak: Tesla can now damage blobs
tweak: Tesla causes machinery to overload and explode when zapping it
tweak: Tesla dissipates energy more slowly and is easier to charge up (making it more deadly when released)
tweak: Tesla will now dust mobs it bumps into
add: Tesla coils now have wires that can be pulsed to generate a tesla zap
/:cl:

### CODE REVIEW

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -46,33 +52,30 @@ var/list/blacklisted_tesla_types = typecacheof(list(/obj/machinery/atmospherics,
 	for(var/ball in orbiting_balls)
 		var/obj/singularity/energy_ball/EB = ball
 		qdel(EB)
-		
-	poi_list.Remove(src)
```
Comment (Fox-McCloud) [inline code/modules/power/tesla/energy_ball.dm]: This is already handled in `singularity/Destroy()`

**Comment by FlattestGuitar** [comment]
>tesla can now damage blobs
I don't see the point there. We'll have five times more releases than ahelps about whether the tesla can be released. Not to mention the fact that people already do that.

>Tesla's energy will dissipate more slowly and charge up more quickly
Does that mean it's even further from being a slow-chargeup engine? I think that was one of the intentions initially.

>tweak: Tesla will not dust mobs it bumps into
CL is wrong, considering what you wrote earlier

>everything else
👌

**Comment by KingPhilipIII** [comment]
Tesla engine releases are terrifying on TG.

**Comment by Fox-McCloud** [comment]
Players already ask admins if they can nuke the station during blob---which is a "kill everyone" kinda thing.

Releasing the tesla is in a similar vein...not to mention the singularity can already eat blobs, and you hardly have everyone and their mother spam asking if they can do that.

The telsa's grid potential tops out at 2.4 million watts, unless you build more coils. 

The buildup was dramatically nerfed a while back--this is buffing it back up a little, while still not having it be the infinite energy engine it used to be---it makes it a more attractive option to release over the singulo; it also means it's not going to die off after a short little bit when it is released.

**Comment by SamHPurp** [comment]
I think this is a good change to make the set-and-forget engine way more deadly.

Will you still be immune to the dusting if you're fully grounded, IE: Engineering void suit and gloves?

**Comment by KingPhilipIII** [comment]
TG recently made a huge change to use only SM engines. Did away with Lord Singuloth and Lady Tesla's containment entirely. Their SM generates a singularity however. Any chance of something similar happening here or no?

**Comment by taukausanake** [comment]
I rather have the semi-old supermatter that generated plasma. Use that plasma in other things like burn chambers or if we didn't have LINDA, a R-Ust engine. That's just me though

**Comment by Fox-McCloud** [comment]
>Will you still be immune to the dusting if you're fully grounded, IE: Engineering void suit and gloves?

No.

**Comment by DesolateG** [comment]
Still going to be the weakest engine (sans Solars and.. ugh.. turbine) sadly.

**Comment by Anticept** [comment]
@Fox-McCloud I'm not sure how to feel about the removal of immunity part. Perhaps the tesla should be brought down to a lower energy state, then engineering can do containment maintenance? As this PR stands right now, once tesla is up, that's it.

Or is the dusting same as always, regarding the core touching you and you getting dusted? Because this is already a thing.

**Comment by Fox-McCloud** [comment]
>Or is the dusting same as always, regarding the core touching you and you getting dusted? Because this is already a thing.

It's the same as it's been, with the exception that if the Tesla bumps into *you*, it will dust you (as opposed to you having to bump into it).

**Comment by tylerh95** [comment]
@Fox-McCloud perhaps leave the immunity for fully grounded people- it's counterplay against the tesla. But should still die if the actual core of the tesla hits you.

**Comment by Fox-McCloud** [comment]
That's what grounding rods are for.

**Comment by SigholtStarsong** [comment]
I like it. Make it so.

**Comment by Allfd** [comment]
Also like,

Super  intrigued by this super matter thing though.

**Comment by KingPhilipIII** [comment]
All hail Lady Tesla.

**Pull request merged**